src/handler-context/Config/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Json } from "../../shared/types";
+import { ConfigClient, ConfigFileNames, ConfigFiles } from "./types";
+
+export class Config<TFileName extends ConfigFileNames> {
+  private readonly _client: ConfigClient;
+  private readonly _files: ConfigFileNames[];
+  private readonly _logger: Logger;
+  private _hasCacheBeenPopulated: boolean = false;
+  private _cache!: Omit<ConfigFiles, keyof Omit<ConfigFiles, TFileName>>;
```

Comment:
yeah, sadly you can't do Pick like this. I guess it isn't able to consider TFileName as a union of strings because it's an extension of an enum or something along those lines.